### PR TITLE
Add a note about zstd incompatibility with k-NN and security analytics

### LINKED ISSUE #5002: [DOC] ZStd Documentation update to exclude knn and correlation indices for v2.10
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 With the moving of ZStd to a plugin in OS v2.10, the users won't be able to use ZStd compression codecs for KNN and Correlation Indices (Security Analytics), since all three plugins implement EnginePlugin and for an index, we can override only one [custom codec service factory method](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/engine/EngineConfigFactory.java#L106)

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/OpenSearch/pull/9658

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #5002 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@sarthakaggarwal97 Could you review this PR please?

**Comment by sarthakaggarwal97** [comment]
LGTM! thanks @kolchfa-aws for taking this up!

File: _im-plugin/index-codecs.md [hunk 1]
```diff
@@ -24,6 +24,9 @@ It may be challenging to change the codec setting of an existing index (see [Cha
 * `zstd` (OpenSearch 2.9 and later) -- This codec provides significant compression comparable to the `best_compression` codec with reasonable CPU usage and improved indexing and search performance compared to the `default` codec.
 * `zstd_no_dict` (OpenSearch 2.9 and later) -- This codec is similar to `zstd` but excludes the dictionary compression feature. It provides faster indexing and search operations compared to `zstd` at the expense of a slightly larger index size.
 
+As of OpenSearch 2.10, the `zstd` and `zstd_no_dict` codecs are not compatible with the [k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/index/) and [Security Analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/) plugins.
```
Comment (sarthakaggarwal97) [inline _im-plugin/index-codecs.md:27]: Can we rephrase it to 


> As of OpenSearch 2.10, the `zstd` and `zstd_no_dict` codecs are not compatible with the [k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/index/) and [Security Analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/) indices.

**Review by sarthakaggarwal97 — CHANGES_REQUESTED** [review]
Minor comment, zstd and zstd_no_dict codecs will not be usable for the knn/ security-analytics indices. They can still be installed together

File: _im-plugin/index-codecs.md (hunk 1 above)
Comment (kolchfa-aws) [inline _im-plugin/index-codecs.md:27]:
```suggestion
As of OpenSearch 2.10, the `zstd` and `zstd_no_dict` compression codecs cannot be used for [k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/index/) and [Security Analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/) indexes.
```

**Review by sarthakaggarwal97 — APPROVED** [review]
thanks @kolchfa-aws , looks good!

File: _im-plugin/index-codecs.md [hunk 2]
```diff
@@ -24,6 +24,9 @@ It may be challenging to change the codec setting of an existing index (see [Cha
 * `zstd` (OpenSearch 2.9 and later) -- This codec provides significant compression comparable to the `best_compression` codec with reasonable CPU usage and improved indexing and search performance compared to the `default` codec.
 * `zstd_no_dict` (OpenSearch 2.9 and later) -- This codec is similar to `zstd` but excludes the dictionary compression feature. It provides faster indexing and search operations compared to `zstd` at the expense of a slightly larger index size.
 
+As of OpenSearch 2.10, the `zstd` and `zstd_no_dict` compression codecs cannot be used for [k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/index/) and [Security Analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/) indexes.
```
Comment (natebower) [inline _im-plugin/index-codecs.md:27]:
```suggestion
As of OpenSearch 2.10, the `zstd` and `zstd_no_dict` compression codecs cannot be used for [k-NN]({{site.url}}{{site.baseurl}}/search-plugins/knn/index/) or [Security Analytics]({{site.url}}{{site.baseurl}}/security-analytics/index/) indexes.
```

**Review by natebower — APPROVED** [review]
LGTM with one minor change.
